pkg/apis/core/v1/conversion_test.go

```diff
@@ -346,3 +346,279 @@ func roundTripRS(t *testing.T, rs *apps.ReplicaSet) *apps.ReplicaSet {
+			},
+		},
+		{
+			// no ips
+			PodIPs: nil,
+		},
+		{
+			// list of ips
+			PodIPs: []core.PodIP{
+				{
+					IP: "1.1.1.1",
+				},
+				{
+					IP: "2000::",
+				},
+			},
+		},
+	}
+	for i, input := range testInputs {
+		v1PodStatus := v1.PodStatus{}
+		if err := corev1.Convert_core_PodStatus_To_v1_PodStatus(&input, &v1PodStatus, nil); nil != err {
+			t.Errorf("%v: Convert core.PodStatus to v1.PodStatus failed with error %v", i, err.Error())
+		}
+
+		if len(input.PodIPs) == 0 {
+			// no more work needed
+			continue
+		}
+		// Primary IP was not set..
+		if len(v1PodStatus.PodIP) == 0 {
+			t.Errorf("%v: Convert core.PodStatus to v1.PodStatus failed out.PodIP is empty, should be %v", i, v1PodStatus.PodIP)
+		}
+
+		// Primary should always == in.PodIPs[0].IP
+		if len(input.PodIPs) > 0 && v1PodStatus.PodIP != input.PodIPs[0].IP {
+			t.Errorf("%v: Convert core.PodStatus to v1.PodStatus failed out.PodIP != in.PodIP[0].IP expected %v found %v", i, input.PodIPs[0].IP, v1PodStatus.PodIP)
+		}
+		// match v1.PodIPs to core.PodIPs
+		for idx := range input.PodIPs {
+			if v1PodStatus.PodIPs[idx].IP != input.PodIPs[idx].IP {
+				t.Errorf("%v: Convert core.PodStatus to v1.PodStatus failed. Expected v1.PodStatus[%v]=%v but found %v", i, idx, input.PodIPs[idx].IP, v1PodStatus.PodIPs[idx].IP)
+			}
+		}
+	}
+}
+func Test_v1_PodStatus_to_core_PodStatus(t *testing.T) {
+	// fail
+	v1FailTestInputs := []v1.PodStatus{
+		{
+			PodIP: "1.1.2.1", // fail becaue PodIP != PodIPs[0]
+			PodIPs: []v1.PodIP{
+				{IP: "1.1.1.1"},
+				{IP: "2.2.2.2"},
+			},
+		},
+	}
+	// success
+	v1TestInputs := []v1.PodStatus{
+		{
+			// both are not provided
+			PodIP:  "",
+			PodIPs: nil,
+		},
+		// only list of IPs
+		{
+			PodIPs: []v1.PodIP{
+				{IP: "1.1.1.1"},
+				{IP: "2.2.2.2"},
+			},
+		},
+		// only Primary IP Provided
+		{
+			PodIP: "1.1.1.1",
+		},
+		// Both
+		{
+			PodIP: "1.1.1.1",
+			PodIPs: []v1.PodIP{
+				{IP: "1.1.1.1"},
+				{IP: "2.2.2.2"},
+			},
+		},
+		// v4 and v6
+		{
+			PodIP: "1.1.1.1",
+			PodIPs: []v1.PodIP{
+				{IP: "1.1.1.1"},
+				{IP: "::1"},
+			},
+		},
+		// v6 and v4
+		{
+			PodIP: "::1",
+			PodIPs: []v1.PodIP{
+				{IP: "::1"},
+				{IP: "1.1.1.1"},
+			},
+		},
+	}
+	// run failed cases
+	for i, failedTest := range v1FailTestInputs {
+		corePodStatus := core.PodStatus{}
+		// convert..
+		if err := corev1.Convert_v1_PodStatus_To_core_PodStatus(&failedTest, &corePodStatus, nil); err == nil {
+			t.Errorf("%v: Convert v1.PodStatus to core.PodStatus should have failed for input %+v", i, failedTest)
+		}
+	}
+
+	// run ok cases
+	for i, testInput := range v1TestInputs {
+		corePodStatus := core.PodStatus{}
+		// convert..
+		if err := corev1.Convert_v1_PodStatus_To_core_PodStatus(&testInput, &corePodStatus, nil); err != nil {
+			t.Errorf("%v: Convert v1.PodStatus to core.PodStatus failed with error:%v for input %+v", i, err.Error(), testInput)
+		}
+
+		if len(testInput.PodIP) == 0 && len(testInput.PodIP) == 0 {
```

Comment:
shouldnt this be `len(testInput.PodIP) == 0 && len(testInput.PodIPs) == 0` rather than the above